### ISSUE TITLE
How to use single GPU for tf?

### ISSUE THREAD

**Issue by mingqizhang** [NVIDIA/DALI#1474]
I have four GPUs in my device, but I only want to use one GPU during training.
When I use following code, four GPUs all create the memory.  Why I use tf.device('/gpu:0') is ineffective?

    daliop = dali_tf.DALIIterator()

    with tf.device('/gpu:0'):
        image, mask = daliop(pipeline = TFRecordPipeline(batch_size=batch_size, num_threads=4, device_id=0, device="gpu"),
                          shapes=[(batch_size, 256, 256, 3), (batch_size, 256,256, 3)],
                          dtypes = [tf.float32, tf.float32],
                          device_id=0)
        gpu_options = tf.GPUOptions(per_process_gpu_memory_fraction=0.1)
        config = tf.ConfigProto(gpu_options=gpu_options)
        with tf.Session(config=config) as sess:
            images, masks = sess.run([image, mask])
`

**Comment by JanuszL** [comment]
Hi,
I believe it is TF thing that it allocates whole GPU memory ahead no matter if you run anything there or not. So it is not the DALI related issue.

**Comment by mingqizhang** [comment]
@JanuszL 
The GPUs status show as follow:
![image](https://user-images.githubusercontent.com/28778038/69054060-c3aae700-0a45-11ea-97fa-22638c48b590.png)
It is normal that use pipe = TFRecordPipeline(batch_size, 4, 0, "gpu");
pipe.build() and the memory is only 200M.
if I not use dali, I can specify the GPU when training. It is confusing for me that why I can not specify the GPU when using dali.
In following tutorials, can it specify the GPU for different pipes?

    import tensorflow as tf
    import nvidia.dali.plugin.tf as dali_tf

    def get_batch_test_dali(batch_size, pipe_type):
        pipe_name, label_type, _ = pipe_type
        pipes = [pipe_name(batch_size=batch_size, num_threads=2, device_id = 
    device_id, num_gpus = N) for device_id in range(N)]

        daliop = dali_tf.DALIIterator()
        images = []
        labels = []
        for d in range(N):
            with tf.device('/gpu:%i' % d):
                image, label = daliop(pipeline = pipes[d],
                    shapes = [(BATCH_SIZE, 3, 227, 227), ()],
                    dtypes = [tf.int32, label_type],
                    device_id = d)
                images.append(image)
                labels.append(label)

        return [images, labels]

**Comment by JanuszL** [comment]
Hi,
[In this example](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/examples/tensorflow/tensorflow-plugin-tutorial.html) DALI pipeline is created for each GPU first and then passed to daliop. If you want to use DALI only for 1 GPU then you need to create the pipeline first only for it.
So you code should look like:
```
class RN50Pipeline(Pipeline):
(...)
DEVICES = 1
pipes = [RN50Pipeline(batch_size=BATCH_SIZE, num_threads=2, device_id = device_id, num_gpus = DEVICES) for device_id in range(DEVICES)]
daliop = dali_tf.DALIIterator()

images = []
labels = []
for d in range(DEVICES):
    with tf.device('/gpu:%i' % d):
        image, label = daliop(pipeline = pipes[d],
            shapes = [(BATCH_SIZE, 3, 224, 224), ()],
            dtypes = [tf.int32, tf.float32])

        images.append(image)
        labels.append(label)
```
If you want to run mult-gpu training take a [look into RN50 example with DALI](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/examples/tensorflow/demo/README.html).
